### PR TITLE
Support api versioning 

### PR DESCRIPTION
**Pull request by MCarlomagno**
This PR allows to call defender 2.0 endpoints  from defender-client.
https://linear.app/openzeppelin-development/issue/PLAT-1665/no-way-to-archive-batch-proposals-with-multiple-contracts

Usage:

```js
  public async someFunction(id: string): Promise<Response> {
    return this.apiCall(async (api) => {
      return await api.put(`suffix/${id}`);
    }, 'v2'); // <- here we specify the version
  });
```

```js
  public async someFunction(id: string): Promise<Response> {
    return this.apiCall(async (api) => {
      return await api.put(`suffix/${id}`);
    }); // <- defaults to 'v1'
  });
```

### CODE REVIEW

File: packages/admin/src/api.ts [hunk 1]
```diff
@@ -160,22 +163,18 @@ export class AdminClient extends BaseApiClient {
...
-  public async unarchiveProposal(contractId: string, proposalId: string): Promise<ProposalResponseWithUrl> {
+  public async unarchiveProposal(_: string, proposalId: string): Promise<ProposalResponseWithUrl> {
     return this.apiCall(async (api) => {
-      const response = (await api.put(`/contracts/${contractId}/proposals/${proposalId}/archived`, {
-        archived: false,
-      })) as ProposalResponse;
+      const response = (await api.put(`proposals/unarchive/${proposalId}`)) as ProposalResponse;
       return { ...response, url: getProposalUrl(response) };
-    });
+    }, 'v2');
```
Comment (MCarlomagno) [inline packages/admin/src/api.ts:177]: We call v2 endpoint for archive/unarchive proposals to fix [this issue](https://github.com/OpenZeppelin/defender-client/issues/263)

File: packages/admin/src/api.ts [hunk 2]
```diff
@@ -160,22 +163,18 @@ export class AdminClient extends BaseApiClient {
...
-        archived: true,
-      })) as ProposalResponse;
+      const response = (await api.put(`proposals/archive/${proposalId}`)) as ProposalResponse;
       return { ...response, url: getProposalUrl(response) };
-    });
+    }, 'v2');
   }
 
-  public async unarchiveProposal(contractId: string, proposalId: string): Promise<ProposalResponseWithUrl> {
+  public async unarchiveProposal(_: string, proposalId: string): Promise<ProposalResponseWithUrl> {
```
Comment (MCarlomagno) [inline packages/admin/src/api.ts:173]: `contractId` is no longer required but we keep the param for backwards compatibility

File: examples/batch-proposal/index.js [hunk 3]
```diff
@@ -71,6 +71,12 @@ async function main() {
   });
 
   console.log(url);
+
+  // archive/unarchive
+  const archived = await client.archiveProposal(contractId, proposalId);
+  console.log('proposal archived', archived.isArchived);
+  const unarchived = await client.unarchiveProposal(contractId, proposalId);
+  console.log('proposal archived', unarchived.isArchived);
```
Comment (MCarlomagno) [inline examples/batch-proposal/index.js:79]: Archive/unarchive added to example for batch proposals

File: packages/admin/src/api.ts [hunk 4]
```diff
@@ -54,7 +54,10 @@ export class AdminClient extends BaseApiClient {
     return process.env.DEFENDER_ADMIN_POOL_CLIENT_ID || '40e58hbc7pktmnp9i26hh5nsav';
   }
 
-  protected getApiUrl(): string {
+  protected getApiUrl(v: ApiVersion): string {
```
Comment (shahnami) [inline packages/admin/src/api.ts:57]: Wondering if it makes sense to make this param optional or default to either version?

**Review by shahnami — APPROVED** [review]
LGTM, however I would consider making the params optional and default to v1.

File: packages/admin/src/api.ts [hunk 5]
```diff
@@ -54,7 +54,10 @@ export class AdminClient extends BaseApiClient {
     return process.env.DEFENDER_ADMIN_POOL_CLIENT_ID || '40e58hbc7pktmnp9i26hh5nsav';
   }
 
-  protected getApiUrl(): string {
+  protected getApiUrl(v: ApiVersion = 'v1'): string {
+    if (v === 'v2') {
+      return process.env.DEFENDER_API_URL_V2 || 'https://defender-api.openzeppelin.com/v2/';
```
Comment (zeljkoX) [inline packages/admin/src/api.ts:59]: nit: `DEFENDER_V2_API_URL` looks little better.

**Review by zeljkoX — APPROVED** [review]
LGTM

Thanks!

**Pull request merged**